crates/evm/src/evm/executor.rs

```diff
@@ -70,14 +88,24 @@ pub(crate) fn execute_multiple_tx<
     }
 
     let block_gas_limit = block_env.gas_limit;
+
     let mut cumulative_gas_used = prev_gas_used;
 
     let mut evm = CitreaEvm::new(db, block_env, config_env, ext);
 
     let mut tx_results = Vec::with_capacity(txs.len());
     for tx in txs {
         let block_available_gas = block_gas_limit - cumulative_gas_used;
-        let result = if tx.transaction.gas_limit() > block_available_gas {
+        let result_and_state = match evm.transact(tx) {
+            Ok(result_and_state) => result_and_state,
+            Err(e) => {
+                tx_results.push(Err(e));
+                continue;
+            }
+        };
+
+        // Check if the transaction used more gas than the available block gas limit
+        let result = if result_and_state.result.gas_used() > block_available_gas {
```

Comment:
let's refactor here to:

cumulative_gas_used + tx.result.gas_used() > block_gas_limit

easier to read imo